MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java

```diff
@@ -4767,6 +4773,22 @@ public static Expression convert(@NonNull JsonArray jsonArray) {
       final List<Expression> arguments = new ArrayList<>();
       if (operator.equals("within")) {
         return within(Polygon.fromJson(jsonArray.get(1).toString()));
+      } else if (operator.equals("distance")) {
+        final String type = jsonArray.get(1).getAsJsonObject().get("type").getAsString();
+        String json = jsonArray.get(1).toString();
+        if ("Point".equals(type)) {
+          return distance(Point.fromJson(json));
+        } else if ("LineString".equals(type)) {
+          return distance(LineString.fromJson(json));
+        } else if ("MultiPoint".equals(type)) {
+          return distance(MultiPoint.fromJson(json));
+        } else if ("Polygon".equals(type)) {
+          return distance(Polygon.fromJson(json));
+        } else if ("MultiPolygon".equals(type)) {
+          return distance(MultiPolygon.fromJson(json));
+        } else {
+          return distance(MultiLineString.fromJson(json));
```

Comment:
We are doing a lot of manual type checking and this logic should be handle, if possible, by the geojson library itself. 
Have you tried using `GeometryGeoJson.fromJson()`?